rascal-lsp/src/main/rascal/demo/lang/pico/LanguageServer.rsc

```diff
@@ -71,10 +71,12 @@ data Command
 rel[loc,Command] picoLenses(start[Program] input) = {<input@\loc, renameAtoB(input, title="Rename variables a to b.")>};
 
 
+loc atEnd(loc l) = l[begin = l.end][offset=l.offset+l.length - 1][length = 1];
+
 list[InlayHint] picoHinter(start[Program] input) {
     typeLookup = ( "<name>" : "<tp>" | /(IdType)`<Id name> : <Type tp>` := input);
     return [
-        hint(name.src, ": <typeLookup["<name>"]>", \type()) | /(Expression)`<Id name>` := input
+        hint(atEnd(name.src), " : <typeLookup["<name>"]>", \type()) | /(Expression)`<Id name>` := input, "<name>" in typeLookup
```

Comment:
see below. I'd like to see `name.src` without the `atEnd` wrapper.

Reply:
Okay, so for now the design is close to LSP's inlayhint feature. Where the start of the source location determines where the hint it put. We could add a feature where you can pass in a location, and a special bool to say: take the end of this location, not the beginning of this location to place this hint?

Reply:
ok